### PR TITLE
Fix: publication in Jenkins and unexpected build behaviors

### PR DESCRIPTION
**Pull request by themr0c**
## What does this pull request change?

Fix publication

* Add memory limits/requests for the publication in Jenkins, and for the podman mockup of Jenkins behavior.
* Allow publication from following branches: main, master, publication (publication to unblock the situation rapidly).
* Disabling url on development playbook to avoid htmltest crawling the live website during PR checks.
* Don't build an archive of the site.
* Fix broken link for devfile docs.
* Fix antora error on assembly_eclipse-che4z.adoc 
* Fix supplemental UI template to avoid htmltest crawling the live website during PR checks.
* Add a step in publication to run htmltest, and avoid publishing an empty site.
* Consider https://projectlombok.org/ as unreliable and skip it during htmltest runs.

## What issues does this pull request fix or reference?

Jenkins build failure leading to publication of an empty website.
https://ci.eclipse.org/che/job/che-docs-pipeline/job/master/471/consoleFull

```
+ ./tools/publication.sh
+ CI=true antora generate antora-playbook-for-publication.yml --stacktrace
Start page specified for site not found: che-7:overview:introduction-to-eclipse-che.adoc
```

This error message happens when you try to build a nonexistent branch.

Jenkins is working hard to have a working copy with minimal content, but we need to fetch the branches. So let's add  these steps:

```
 git config --add remote.origin.fetch +refs/heads/*:refs/remotes/origin/*
  git fetch
```
Jenkins working hard  not to fetch branches:

```
[Pipeline] checkout

The recommended git tool is: git

using credential github-bot
Cloning the remote Git repository
Cloning with configured refspecs honoured and without tags

Cloning repository https://github.com/eclipse/che-docs.git
 > git init /home/jenkins/agent/workspace/che-docs-pipeline_PR-2102/che-docs # timeout=10
Fetching upstream changes from https://github.com/eclipse/che-docs.git
 > git --version # timeout=10
 > git --version # 'git version 2.20.1'
using GIT_ASKPASS to set credentials GitHub bot (username/token)
 > git fetch --no-tags --force --progress -- https://github.com/eclipse/che-docs.git +refs/pull/2102/head:refs/remotes/origin/PR-2102 +refs/heads/master:refs/remotes/origin/master # timeout=10

Fetching without tags

Merging remotes/origin/master commit 4839d731d9545c3608129bd2819230bea8c37c0e into PR head commit 337428e48a5ccff2374d3f0f94e715c0c37fb726
 > git config remote.origin.url https://github.com/eclipse/che-docs.git # timeout=10
 > git config --add remote.origin.fetch +refs/pull/2102/head:refs/remotes/origin/PR-2102 # timeout=10
 > git config --add remote.origin.fetch +refs/heads/master:refs/remotes/origin/master # timeout=10
 > git config remote.origin.url https://github.com/eclipse/che-docs.git # timeout=10
Fetching upstream changes from https://github.com/eclipse/che-docs.git
using GIT_ASKPASS to set credentials GitHub bot (username/token)
 > git fetch --no-tags --force --progress -- https://github.com/eclipse/che-docs.git +refs/pull/2102/head:refs/remotes/origin/PR-2102 +refs/heads/master:refs/remotes/origin/master # timeout=10
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 337428e48a5ccff2374d3f0f94e715c0c37fb726 # timeout=10
Merge succeeded, producing 337428e48a5ccff2374d3f0f94e715c0c37fb726
Checking out Revision 337428e48a5ccff2374d3f0f94e715c0c37fb726 (PR-2102)

 > git remote # timeout=10
 > git config --get remote.origin.url # timeout=10
using GIT_ASKPASS to set credentials GitHub bot (username/token)
 > git merge 4839d731d9545c3608129bd2819230bea8c37c0e # timeout=10
 > git rev-parse HEAD^{commit} # timeout=10
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 337428e48a5ccff2374d3f0f94e715c0c37fb726 # timeout=10

Commit message: "Fix publication"
First time build. Skipping changelog.
```

## Pull Request checklist

The author and the reviewers validate the content of this pull request with the following checklist, in addition to the [automated tests](code_review_checklist.adoc).

- Any procedure:
  - [ ] Successfully tested.
- Any page or link rename:
  - [ ] The page contains a redirection for the previous URL.
  - Propagate the URL change in:
    - [ ] Dashboard [default branding data](https://github.com/eclipse-che/che-dashboard/blob/main/packages/dashboard-frontend/src/services/bootstrap/branding.constant.ts)
    - [ ] Chectl [constants.ts](https://github.com/che-incubator/chectl/blob/master/src/constants.ts)
- [ ] Builds on [Eclipse Che hosted by Red Hat](https://workspaces.openshift.com).
- [ ] the *`Validate language on files added or modified`* step reports no vale warnings.

### CODE REVIEW

**Review by rkratky — APPROVED** [review]

File: Jenkinsfile [hunk 1]
```diff
@@ -52,7 +52,11 @@ spec:
 
     stage('Checkout www repo (master)') {
       when {
-        branch 'master'
+          anyOf {
+            branch 'master';
```
Comment (ibuziuk) [inline Jenkinsfile:71]: is it planned to switch to the 'main' branch?

File: tools/publication.sh [hunk 2]
```diff
@@ -16,5 +16,6 @@ set -ex
 # They should be run by make-release.sh when creating or updating the 7.y.x branch.
 #./tools/environment_docs_gen.sh
 #./tools/checluster_docs_gen.sh
+git fetch
 
 CI=true antora generate antora-playbook-for-publication.yml --stacktrace
```
Comment (ibuziuk) [inline tools/publication.sh]: no new line

File: Jenkinsfile (hunk 1 above)
Comment (themr0c) [reply Jenkinsfile:71]: Yes, but later.

**Review by l0rd — APPROVED** [review]

**Comment by themr0c** [comment]
Link checker testing the live website and not the built website ? WTF ?  I never thought of that.

File: Jenkinsfile (hunk 1 above)
Comment (themr0c) [reply Jenkinsfile:71]: Let's get ready for it.

**Comment by themr0c** [comment]
Let's summarize the findings and the changes:


-  Add memory limits/requests for the publication in Jenkins, and for the podman mockup of Jenkins behavior.
-  Allow publication from following branches: main, master, publication (publication to unblock the situation rapidly).
-  Disabling url on development playbook to avoid htmltest crawling the live website during PR checks.
-  Don't build an archive of the site.
-  Fix broken link for devfile docs.
-  Fix antora error on assembly_eclipse-che4z.adoc
-  Fix supplemental UI template to avoid htmltest crawling the live website during PR checks.
-  Add a step in publication to run htmltest, and avoid publishing an empty site.

File: tools/publication.sh (hunk 2 above)
Comment (themr0c) [reply tools/publication.sh]: fixed

**Pull request merged**